build-system/compile/pre-closure-babel.js

```diff
@@ -60,26 +57,48 @@ function getFilesToTransform() {
  * @return {!Promise}
  */
 function preClosureBabel() {
+  const babelPlugins = conf.plugins({
+    isForTesting: !!argv.fortesting,
+    isEsmBuild: !!argv.esm,
+    isSinglePass: !!argv.single_pass,
+    isChecktypes: argv._.includes('check-types'),
+  });
+  const babel = gulpBabel({
+    plugins: babelPlugins,
+    retainLines: true,
+    compact: false,
+  });
+
   return through.obj((file, enc, next) => {
-    const cachedTransform = cachedTransforms[file.path];
+    const {relative, path} = file;
+    if (!filesToTransform.includes(relative)) {
+      return next(null, file);
+    }
+
+    const cachedTransform = cachedTransforms[path];
```

Comment:
Sorry @jridgewell, I just realized there's a potential bug in this line of code (written by me).

When we start using `gulp-watch` to incrementally re-compile files that have been edited (WIP branch [here](https://github.com/ampproject/amphtml/compare/master...rsimha:2020-03-26-LazyBuildMinified)), using the filename as the key will cause old contents to be returned. Instead, we want to re-run transforms on the file that was edited.

One way to fix this is to pass in the entry point file to `preClosureBabel()` and always re-transform that file (while using the cache for other files).

Is this something you can add to this PR? (I can fix in a follow up if it's too complicated).

Reply:
I added a contents check to support this, so we wouldn't have to plumb through extra items.

Reply:
Fantastic!